src/webui/src/components/PackageSidebar/index.jsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import API from '../../../utils/api';
+
+import classes from './style.scss';
+import getRegistryURL from '../../../utils/getRegistryURL';
+
+export default class PackageSidebar extends React.Component {
+  state = {
+    lastUpdate: 'Loading',
+    recentReleases: [],
+    author: null,
+    contributors: null,
+    showAllContributors: false,
+    dependencies: {}
+  };
+
+  constructor(props) {
+    super(props);
+    this.showAllContributors = this.showAllContributors.bind(this);
+  }
+
+  async componentDidMount() {
+    await this.loadPackageData(this.props.packageName);
+  }
+
+  async componentWillReceiveProps(newProps) {
+    if (newProps.packageName !== this.props.packageName) {
+      await this.loadPackageData(newProps.packageName);
+    }
+  }
+
+  async loadPackageData(packageName) {
+    let packageMeta;
```

Comment:
API calls can be separated . It would be easy to test and will look cleaner.

Reply:
Yeah, @ayusharma  is right. I think we might create an API utilities and define there all endpoints.